libs/librrgraph/src/io/rr_graph_reader.cpp

```diff
@@ -115,3 +134,140 @@ void load_rr_file(RRGraphBuilder* rr_graph_builder,
             read_rr_graph_name);
     }
 }
+
+static RREdgeId process_rr_edge_override(const std::string& line,
+                                         std::vector<float>& overridden_values,
+                                         const RRGraphView& rr_graph) {
+    std::istringstream iss(line);
+    char ch;
+    RREdgeId edge_id;
+
+    if (std::isdigit(line[0])) {
+        // Line starts with an integer
+        int first;
+        iss >> first;
+        edge_id = (RREdgeId)first;
+    } else if (line[0] == '(') {
+        // Line starts with (first, second)
+        int first, second;
+        iss >> ch >> first >> ch >> second >> ch;
+
+        RRNodeId src_node_id = RRNodeId(first);
+        RRNodeId sink_node_id = RRNodeId(second);
+
+        for (RREdgeId outgoing_edge_id : rr_graph.rr_nodes().edge_range(src_node_id)) {
+            if (rr_graph.rr_nodes().edge_sink_node(outgoing_edge_id) == sink_node_id) {
+                edge_id = outgoing_edge_id;
+                break;
+            }
+        }
+
+        if (!edge_id.is_valid()) {
+            VTR_LOG_ERROR("Couldn't find an edge connecting node %d to node %d\n", src_node_id, sink_node_id);
+        }
+
+    } else {
+        VTR_LOG_ERROR("Invalid line format:  %s\n", line.c_str());
+    }
+
+    float value;
+    while (iss >> value) {
+        overridden_values.push_back(value);
+    }
+
+    return edge_id;
+}
+
+struct t_rr_switch_inf_hash {
```

Comment:
I think a better way to implement a hash function for `t_rr_switch_inf` is to define the hash function within the struct itself. This ensures that if someone adds a new field to the data structure, they can update the hash function directly in the same place.  

Additionally, you can add the following specialization in the same file:  
```cpp
namespace std {
    template <>
    struct hash<t_rr_switch_inf> {
        size_t operator()(const t_rr_switch_inf& s) const {
            return s.hash();
        }
    };
}
```
An added benefit of this approach is that you wouldn't need to explicitly pass the hash function when using data structures like `std::unordered_map` or `std::unordered_set`, as it will be automatically recognized.

Reply:
I kind of like @amin1377's suggestion, although @soheilshahrouz tells me he's less keen.  I like the simplicity of having a hash function that gets used automatically where appropriate.